tests/handlers/test_checkers.py

```diff
@@ -0,0 +1,126 @@
+from typing import Dict
+from unittest.mock import AsyncMock, Mock
+
+from twisted.internet.defer import ensureDeferred
+from twisted.test.proto_helpers import MemoryReactor
+from twisted.trial import unittest
+from twisted.web.client import PartialDownloadError
+
+from synapse.api.errors import Codes, LoginError
+from synapse.handlers.ui_auth.checkers import RecaptchaAuthChecker
+from synapse.server import HomeServer
+from synapse.util import json_encoder
+
+
+class TestRecaptchaAuthChecker(unittest.TestCase):
+    def setUp(self: "TestRecaptchaAuthChecker") -> None:
+        self.hs = Mock(spec=HomeServer)
+        self.hs.config = Mock()
+        self.hs.config.captcha.recaptcha_private_key = "test_private_key"
+        self.hs.config.captcha.recaptcha_siteverify_api = (
+            "https://www.recaptcha.net/recaptcha/api/siteverify"
+        )
+        self.http_client = AsyncMock()
+        self.hs.get_proxied_http_client.return_value = self.http_client
+        self.recaptcha_checker = RecaptchaAuthChecker(self.hs)
+        self.reactor = MemoryReactor()
+
+    def test_is_enabled(self: "TestRecaptchaAuthChecker") -> None:
+        """Test that the checker is enabled when a private key is configured."""
+        self.assertTrue(self.recaptcha_checker.is_enabled())
+
+    def test_is_disabled(self: "TestRecaptchaAuthChecker") -> None:
+        """Test that the checker is disabled when no private key is configured."""
+        self.hs.config.captcha.recaptcha_private_key = None
+        recaptcha_checker = RecaptchaAuthChecker(self.hs)
+        self.assertFalse(recaptcha_checker.is_enabled())
+
+    def test_check_auth_success(self: "TestRecaptchaAuthChecker") -> None:
+        """Test that authentication succeeds with a valid recaptcha response."""
+        _expected_response = {"success": True}
+        self.http_client.post_urlencoded_get_json = AsyncMock(
+            return_value=_expected_response
+        )
+
+        authdict = {"response": "captcha_solution", "session": "fake_session_id"}
+        clientip = "127.0.0.1"
+
+        d = ensureDeferred(self.recaptcha_checker.check_auth(authdict, clientip))
+
+        result = self.successResultOf(d)
+        self.assertTrue(result)
```

Comment:
This and other areas can be simplified using the `HomeserverTestCase.get_success` method. It will await on the deferred for you, and return the result.

```suggestion
        user_authed = self.get_success(self.recaptcha_checker.check_auth(authdict, clientip))
        self.assertTrue(user_authed)
```